File: C++/50. Pow(x, n).cpp
class Solution {
public:
    double myPow(double x, int n) {
        double res = 1;
        long long tmp = n;
        if(tmp < 0)
        {
            tmp = -tmp;
            x = 1.0/x;
        }
        while(tmp > 0)
        {
            if(tmp%2!=0)
                res = res*x;
            x = x*x;
            tmp >>= 1;
        }
        return res;
    }
};

File: Python/50. Pow(x, n).py
class Solution:
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        res = 1
        if n < 0:
            n = -n
            x = 1.0/x
        while n > 0:
            if n % 2 != 0:
                res = res * x
            x = x * x
            n >>= 1
        return res